rclpy/test/test_wait_set.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2017 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+from rcl_interfaces.srv import GetParameters
+import rclpy
+from rclpy.impl.implementation_singleton import rclpy_implementation as _rclpy
+from rclpy.wait_set import WaitSet
+from std_msgs.msg import String
+
+
+class TestWaitSet(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        rclpy.init()
+        cls.node = rclpy.create_node('TestWaitSet', namespace='/')
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.node.destroy_node()
+        rclpy.shutdown()
+
+    def test_guard_condition_ready(self):
+        ws = WaitSet()
+        gc_handle, gc_pointer = _rclpy.rclpy_create_guard_condition()
+        try:
+            ws.add_guard_condition(gc_handle, gc_pointer)
+            self.assertFalse(ws.is_ready(gc_pointer))
+
+            ws.wait(1)
+            self.assertFalse(ws.is_ready(gc_pointer))
+
+            _rclpy.rclpy_trigger_guard_condition(gc_handle)
+            # TODO(sloretz) why does the next assertion fail with wait(0)?
```

Comment:
hmmm good question, would be good to find out why

Reply:
@dhood nice find! I hadn't looked into it

Reply:
https://github.com/ros2/rmw_fastrtps/pull/158 _should_ let you change back to `wait(0)`

Reply:
Works on my machine. Thanks @dhood 385b18bfd5e3af11d805c6b66a408336eafc7026